projects/task_list/app.js

```diff
@@ -0,0 +1,169 @@
+// Define UI Vars
+const form = document.querySelector('#task-form');
+const taskList = document.querySelector('.collection');
+const clearBtn = document.querySelector('.clear-tasks');
+const filter = document.querySelector('#filter');
+const taskInput = document.querySelector('#task');
+
+// Load all event listeners
+loadEventListeners();
+
+// Load all event listeners
+function loadEventListeners() {
+    // DOM load event
+    document.addEventListener('DOMContentLoaded', getTasks);
+    // Add task event
+    form.addEventListener('submit', addTask);
+    // Remove task event
+    taskList.addEventListener('click', removeTask);
+    // Clear tasks event
+    clearBtn.addEventListener('click', clearTasks);
+    // Filter tasks event
+    filter.addEventListener('keyup', filterTasks);
+}
+
+// Get tasks from local storage
+function getTasks() {
+    let tasks;
+    if(localStorage.getItem('tasks') === null) {
+        tasks = [];
+    } else {
+        tasks = JSON.parse(localStorage.getItem('tasks'));
+    }
+
+    tasks.forEach(function(task){
+        // Create li element
+        const li = document.createElement('li');
+        // Add class to li
+        li.className = 'collection-item';
+        // Create text node & append to li
+        li.appendChild(document.createTextNode(task));
+        // Create new link element
+        const link = document.createElement('a');
+        // Add class to link
+        link.className = 'delete-item secondary-content';
+        // Add icon HTML
+        link.innerHTML = '<i class="fa fa-remove"></i>';
+        // Append link to li
+        li.appendChild(link);
+
+        // Append li to ul
+        taskList.appendChild(li);
+    });
+}
+
+// Add task function
+function addTask(e) {
+    if(taskInput.value === '') {
+        alert('No task added!');
+    }
+
```

Comment:
To your comment and our discussion, try to reason about this area and pretend to roll through the code yourself. I think you'll understand why we're able to add empty tasks.

Reply:
Wow that was really simple. No else statement at the end of the if so Javascript just carried on with the following code as it was just new instructions and the if statement had be successfully passed. I think i briefly thought about him missing that when I was coding along but didn't think much of it because the product was still functional.

Reply:
Carrying the learning from that mistake, I have chosen to put the closing curly brace before the e.preventDefault() so that instruction is carried out in both the case where the field is populated and empty. 

That was satisfying to think through even though it was utterly simple 😂

Reply:
Pretty classic knowing that bug is in his code in the code through but undiscovered